File: DI/DI-Dagger-Java/src/main/java/DI/persona/modules/PersonasRepositoryModule.java
package DI.persona.modules;

import dagger.Binds;
import dagger.Module;
import model.persona.repositories.IPersonaRepository;
import model.persona.repositories.PersonaRepository;

import javax.inject.Singleton;

@Module
public interface PersonasRepositoryModule {
    @Singleton
    @Binds
        // No devuelvo nada, solo bindeo el objeto,por eso no privider
    IPersonaRepository bindPersonaRepository(PersonaRepository impl);
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/DI/personas/modules/PersonasRepositoryModule.kt
package DI.personas.modules

import dagger.Binds
import dagger.Module
import models.personas.repositories.IPersonaRepository
import models.personas.repositories.PersonaRepository
import javax.inject.Singleton


@Module
interface PersonasRepositoryModule {
    // No devuelvo nada, solo bindeo el objeto,por eso no privider
    @Singleton
    @Binds
    fun bindPersonaRepository(impl: PersonaRepository): IPersonaRepository
}
